### ISSUE TITLE
New ZIM request: scp-wiki.wikidot.com

### ISSUE THREAD

**Issue by IMayBeABitShy** [openzim/zim-requests#402]
- Website URL:https://scp-wiki.wikidot.com/
- License: Creative Commons Share-Alike 3.0 License (**Important note:** contains a single image not released under CC, but the creator *'allow[s] the use of the image of "Untitled 2004" by the SCP Foundation and its fanbase for non-commercial purposes only.'*, more information [here](https://scp-wiki.wikidot.com/licensing-guide) and [here (near bottom)](https://scp-wiki.wikidot.com/scp-173)). When using `zimit`, it *may* be necessary to specify something like `--exclude "SCP-173\\.jpg"`.
- Desired ZIM Title: **SCP-Wiki**
- Desired ZIM Description: **The SCP-Wiki is a collaborative creative writing website about the fictional SCP Foundation**
- Desired ZIM Icon –png (URL or attach one): https://scp-wiki.wdfiles.com/local--files/main/logo_white.png
- Language (ISO 639-3): **eng**
- Is this a MediaWiki?: **no**

---

As the majority of this website (the exception being the 'random article'-buttons, login functionality and search) does not seem to need any backend whatsoever, downloading it via `zimit` seems like a viable option. Nearly everything on the website is CC-SA, the only exception (?) being the image of SCP-173, but excluding that one should be easy when using `zimit`. I am not even sure if it even *needs* to be excluded.

**Comment by lbrunkho** [comment]
+1 for this. I actually made a clone of this website using httrack about a year ago and it was an ORDEAL! Would much rather this be in a zim file for my kiwix server. On a side-note the image of 173 is going to get a redesign in the near future to avoid this issue.

**Comment by Popolechien** [comment]
> downloading it via zimit seems like a viable option. Nearly everything on the website is CC-SA, the only exception (?) being the image of SCP-173, but excluding that one should be easy when using zimit

@IMayBeABitShy Have you tried using the limited version of zimit already? did it work?

I had a cursory look but cannot see whether this is mediawiki-based or not.

**Comment by IMayBeABitShy** [comment]
Following @Popolechien's suggestion, I've used youzim.it to create a limited ZIM of the site. It seems like the website works (obviously some stuff like the search doesn't, but zim files have their own search functionality anyway). I did, however, noticed that a lot of junk javascript has been included (e.g. cookie confirmation, ...).

I suggest also excluding the following sites:
- ad-delivery.net
- ad.doubleclick.net
- btloader.com (not sure, couldn't find anything about this)
- consent.nit.ro
- s.nitropay.com
- stats.g.doubleclick,net
- tracker.nitropay.com

This list is *probably* incomplete, but this should be the most important ones on the main page.

> I had a cursory look but cannot see whether this is mediawiki-based or not.

I don't think it is. There is a [wikidot -> mediawiki conversion tool](https://github.com/wilbefast/wikidot-to-mediawiki), which also indicates that it's not a media wiki. Still, I only have superficial knowledge of wiki software, so I may be wrong.

**Comment by Popolechien** [comment]
@IMayBeABitShy Awesome, I've started [a recipe](https://farm.openzim.org/recipes/SCP-wiki_en). Let us see what happens.

**Comment by IMayBeABitShy** [comment]
I think this one failed. I've checked the log a couple of times and zimit seemed to spend a lot of time parsing some background pages (like `workbench` I think they were called). The last time I've checked, the job was finally interrupted.

**Comment by lbrunkho** [comment]
Looks like the favicon URL has changed. 
New URL: https://scp-wiki.wikidot.com/local--favicon/favicon.gif
Also, the recipe log is flooded with these errors. I unfortunately am not familiar enough with zimit to know what this means. 

`[2023-07-02 17:23:23,192: WARNING] failed to load progress details: Expecting value: line 1 column 1 (char 0)`

We can also omit the copyright concern with scp-173 image as this has been removed from the site to adhere to CC BY-SA license.

**Comment by lbrunkho** [comment]
Another update to this request, the attempt on December 29, 2023 was successful! The resulting ZIM was usable, however, it looks like the depth needs to be increased by at least one. 

https://farm.openzim.org/pipeline/6cc5755f-e0de-4a4c-a22f-fa9e43a0603f

Articles listed on the homepage are indexed but the majority of articles are under the series page that are just too deep. 

https://scp-wiki.wikidot.com/scp-series

**Comment by lbrunkho** [comment]
@Popolechien can you reopen this issue or update the recipe for this?

**Comment by Popolechien** [comment]
Just so everyone is on the same page the latest version available is at https://dev.library.kiwix.org/viewer#scp-wiki_en_all 

As far as poking at the zimit recipe goes I'll defer to @benoit74

**Comment by benoit74** [comment]
@lbrunkho @MCSeekeri I'm sorry but I don't get what your issues are. 

Can you please provide link to a page with a non-working link (and details about this non-working link, e.g. position on the screen, text, screenshot, ...) so that I can understand what you are speaking about?

**Comment by benoit74** [comment]
If the page returns a 503, unfortunately there is nothing we can do ... But here the message says "Page Crashed on Load", so I suspect there is another issue. Will have a look when time will be available to work on this ZIM request.

**Comment by Popolechien** [comment]
@leberschnitzel Nothing wrong - the file was in dev and got deleted when we got the [Hetzner Incident](https://mastodon.social/@kiwix/113622081750449356). I'll restart the recipe.

**Comment by Popolechien** [comment]
Damn, I got fooled by the wiki part and thought no further /o\

**Comment by leberschnitzel** [comment]
now I'm confused?

**Comment by Popolechien** [comment]
@leberschnitzel We use different tools depending on the target website (a Mediawiki-based website would use MWoffliner; others would need zimit). I assigned the wrong tool to the task, believing that SCP wiki was mediawiki-based (there are other wiki softwares, much like you can edit documents on Word, LibreOffice or Google Docs).
Long story short, I've restarted the previous recipe and you can follow its progress at https://farm.openzim.org/pipeline/0b1f37ad-fd7d-4501-a787-1a5e40c54926

I'd say we'll find out if it worked sometime tonight / tomorrow morning.

**Comment by benoit74** [comment]
I cancelled the task and disabled the recipe, all calls where finishing in timeouts. Unfortunately is the website is not reliable, it is hard to hope being able to create a ZIM ...

**Comment by leberschnitzel** [comment]
I've used the recipe now to try and download a zim but I get the same problem: It runs well at first but after some time every request is a timeout :(

**Comment by Kufat** [comment]
@leberschnitzel Wikidot is unreliable and times out regularly even under normal use, unfortunately. You'll need to tolerate occasional failures and retry. Note that the non-CC image for SCP-173 is gone now so that's no longer a concern. 
(PS, be careful not to flood on IRC; you got auto-kicked.)

**Comment by leberschnitzel** [comment]
> [@leberschnitzel](https://github.com/leberschnitzel) Wikidot is unreliable and times out regularly even under normal use, unfortunately. You'll need to tolerate occasional failures and retry. Note that the non-CC image for SCP-173 is gone now so that's no longer a concern. (PS, be careful not to flood on IRC; you got auto-kicked.)

not just kicked, also banned... I didn't know it counts new lines as new message :(

**Comment by leberschnitzel** [comment]
> > I cancelled the task and disabled the recipe, all calls where finishing in timeouts. Unfortunately is the website is not reliable, it is hard to hope being able to create a ZIM ...
> 
> I've been working on using zimit to scrape scp-wiki-cn, and in zimit version 2.1.3 it successfully generated working ZIM files. However, it seems to have failed for various reasons since then, which is quite strange...
You mentioned also that it worked for you in 2.1.4 [here](https://github.com/openzim/zimit/issues/376#event-16296812123) so I'm going back to that version and see if I can make something with this.

**Comment by leberschnitzel** [comment]
creating the zim with 2.1.4 worked and has the same problem as mentioned MCSeekeri with SCP-2998
but it's already far better than anything I've tried with 3.0.1.

**Comment by Popolechien** [comment]
I'm looking at the latest run from 10 days ago and it looks pretty good to me - but then I don't know what I should be looking for either. Can someone here please check it out and give feeback? https://dev.library.kiwix.org/viewer#scp-wiki_en_all

**Comment by leberschnitzel** [comment]
> I'm looking at the latest run from 10 days ago and it looks pretty good to me - but then I don't know what I should be looking for either. Can someone here please check it out and give feeback? https://dev.library.kiwix.org/#lang=eng&q=scp&category=
> 
> Ninja edit: fixed link

The much smaller size than my copy (2 vs 20 GB) makes me think that there's lots missing.
If you go to "SCP by Series" on the left side and open any of the roman numerals, and in there open any SCP, it seems to open an outside link.

**Comment by benoit74** [comment]
The last run is limited to 1000 pages, so there is obviously a lot of things missing.

I want to analyze the result to understand what is going on a bit better, didn't had time to dig into it yet.

**Comment by shadow-absorber** [comment]
how far along has this come since mar 13??? the one on dev.library.kiwix.org still links out to the wikidot site instead of being self contained... anyone got new updates???

**Comment by shadow-absorber** [comment]
seems that the zimit docker now works for making an scp wiki zim file but the recipe on [https://farm.openzim.org](https://farm.openzim.org) needs updating to have a higher page count or no limit... here is the command shadow is using to try and build one with all the pages included that they modified from the recipe config:
```
docker run -v $(pwd)/output:/output:rw --name zimit_scp-wiki_en --detach --cpu-shares 3072 --memory 4294967296 --shm-size 4294967296 ghcr.io/openzim/zimit:3.0.5 zimit --seeds=https://scp-wiki.wikidot.com --title='SCP Foundation wiki' --description='A collaborative creative writing website about the (fictional) SCP Foundation' --favicon=https://upload.wikimedia.org/wikipedia/commons/thumb/e/ec/SCP_Foundation_%28emblem%29.svg/512px-SCP_Foundation_%28emblem%29.svg.png --publisher=openZIM --mobileDevice='Pixel 2' --failOnFailedSeed --failOnInvalidStatus --verbose --keep --output=/output --adminEmail=contact+zimfarm@kiwix.org --zim-lang=eng --zimit-progress-file=/output/task_progress.json --name=scp-wiki_en_all
```

**Comment by shadow-absorber** [comment]
> seems that the zimit docker now works for making an scp wiki zim file but the recipe on https://farm.openzim.org needs updating to have a higher page count or no limit... here is the command shadow is using to try and build one with all the pages included that they modified from the recipe config:
> 
> ```
> docker run -v $(pwd)/output:/output:rw --name zimit_scp-wiki_en --detach --cpu-shares 3072 --memory 4294967296 --shm-size 4294967296 ghcr.io/openzim/zimit:3.0.5 zimit --seeds=https://scp-wiki.wikidot.com --title='SCP Foundation wiki' --description='A collaborative creative writing website about the (fictional) SCP Foundation' --favicon=https://upload.wikimedia.org/wikipedia/commons/thumb/e/ec/SCP_Foundation_%28emblem%29.svg/512px-SCP_Foundation_%28emblem%29.svg.png --publisher=openZIM --mobileDevice='Pixel 2' --failOnFailedSeed --failOnInvalidStatus --verbose --keep --output=/output --adminEmail=contact+zimfarm@kiwix.org --zim-lang=eng --zimit-progress-file=/output/task_progress.json --name=scp-wiki_en_all
> ```

this kinda works... takes a lot of time and something causes it to crash after getting around 45k ish pages and restarting more then once after that point seems to cause issues

**Comment by LTVA1** [comment]
I wonder if we can also get all the forum pages, I tried some in the past but new forum pages (beyond 1st one) are loaded by some tricky JS that requires some custom browsertix rule?? to click through them all. I may try to crawl it this weekend. Does anybody know how many days it would take? I see mention of 20GB file but I can't have more than around 60 hours of uninterrupted crawl unless I buy a dedicated laptop.

The motivation for forum pages is that article authors tend to post the explanations of their works there, references and riddles and such.

**Comment by observeroftime02** [comment]
I would have requested this site but saw there was already an open request here. I'm interested in this as well.

I downloaded the file https://mirror.download.kiwix.org/zim/.hidden/dev/scp-wiki_en_all_2025-11.zim and loaded it. I can browse it fine, but the "main meat" on the pages still links out to the internet (the actual SCP wiki entries), which is what I mostly wanted this for.

Are there any problems that need solving, or can I make myself useful in some other small way? I got bandwidth, storage, and (reasonable) compute power aplenty.

**Comment by Popolechien** [comment]
@observeroftime02 The ZIM is not ready for download, as you can see from the discussion and the `needs fixing` tag in the recipe funnel. Unfortunately this is a scraper issue rather than simple compute (but thanks for the offer, you can make yourself known at hello@kiwix.org if you'd like to offer resources - do also check out https://github.com/openzim/zimfarm/blob/main/worker/README.md#requirements for technical requirements). 

We'll be looking at fixing the recipe over the next few weeks (as other priorities permit).

**Comment by MCSeekeri** [comment]
I'm also working on archiving this website, but for some unknown reason, zimit always seems to enter a zombie state after crawling a certain number of pages, or it exits during the process. I'm still looking for a solution.

**Comment by benoit74** [comment]
Just restarted recipe with most recent zimit version, let's see how it goes

**Comment by LTVA1** [comment]
There may be a difficulty with downloading pages' discussions and other forum threads because the 2nd, 3rd, etc. pages are fetched not by direct links but by some Javascript event triggered when you press next page or the number of the page of the particular thread. Idk if crawler could be customized to capture it all, but in my previous attempts it did manage to save only first pages of each thread

**Comment by IMayBeABitShy** [comment]
The recipe [completed succesffully](https://farm.openzim.org/pipeline/15b331ce-963a-4729-b9ab-ed4487059b70). The wiki itself looks mostly great, but I encounterd two minor problems:

1. the icon seems to not be working. Using `kiwix-serve`, it displays as a white image in the viewer.
2. When using the random article button, I get a blank page quite often. An example URL: `http://localhost:8080/viewer#scp-wiki_en_all_2026-03/interwiki.scpwiki.com/interwikiFrame.html%3Flang%3Den%26community%3Dscp%26pagename%3Dflotsam`. These pages seem to be caused by an exception raised by Crom, telling the user that a wikidot.com or scpfoundation.net branch URL is required. Perhaps these simply can't be offlined. Though, in that case they'd ideally not be marked as front articles.

Anyway, for me this would be more than good enough. Thank you all for your support.

**Comment by observeroftime02** [comment]
I downloaded the most recent zim and tried it out on my internet in a box installation. It's looking way better than any previous attempts. For starters, the actual SCP entries are now properly in the zim and all the weird javascript navigation within those pages appears to work and loads the entry's contents. Previous attempts has had the SCP articles (scp-wiki.wikidot.com/scp-002 etc.) linking out to wikidot, which completely defeated the point IMHO. I spent probably 3 hours last night poking around in the archives, looking for any breakage. The forums still link out to wikidot and are not part of the archive it seems, but this is still far better than any previous tries, content wise. All of the main entries are there, I clicked through approximately 100 of them and did not notice anything breaking or linking out.

I am really happy and grateful to have a mostly working offline archive of this, it's huge. In this current state I'm happy enough. It would be nice to also have the forums archived as some posts there contain more information and background stories, as well as discussions, and it's neat to have for universe lore. But, I understand that offlining this is a bit of a PITA (I tried it myself and gotten nowhere). Could the forums be archived as a separate entity? Maybe strictly having an archive for the wiki (the "meat" of the site) and one for the forums (the "flavor") could make this a bit simpler, idk?

In any case, a huge thanks to @benoit74 and anyone else that has spent time on this. I'm holding on to this archive and never letting go, haha.

**Comment by LTVA1** [comment]
Try SCP-4558 — the crawl didn't catch any articles using ListPages Wikidot functionality. You can only see the first offset (first page of the article, if you click "go to SCP-4558 terminal" at the bottom you will be prompted with external resource dialog) but not the next ones

**Comment by LTVA1** [comment]
Forum wasn't saved completely, even the first pages of discussions

**Comment by observeroftime02** [comment]
@LTVA1 good catch, those articles seem to be broken. Is there anything that can be done about it? Is there an easy way of identifying which pages are affected?

**Comment by LTVA1** [comment]
I don't know, you need to increase crawling depth to maaaybe catch e.g. next pages of SCP-4558 (and there are idk at least 20 more articles that also use listpages) but idk what to do with forum, as I already wrote, you need to simulate button click events to fetch any page of any forum thread beyond first one. Maybe you can take a completely different route — try to set up local Wikidot server and ask SCP site admins for a database of the site? (excluding private info like users' passwords etc.) Then you would be able to crawl it fully locally and e.g. set up browsertix custom actions to actually fetch the forum and other stuff. But that's too complicated and just extending crawl depth will make crawling way longer

**Comment by LTVA1** [comment]
Besides there are some very complicated interactive articles that may fetch additional resources only after specific interaction so they can't be archived properly, that's a real pain. Fortunately there aren't that many afaik. I am sort of out of touch with recent SCP site activity from around 2022 when I switched to different hobbies

**Comment by Popolechien** [comment]
Moving to PROD then - unless @benoit74 you want to try something else / more. LMK

**Comment by benoit74** [comment]
I don't think this ZIM passes Kiwix quality expectations.

From my PoV, we need to fix:
- the illustration
- the random URL functionality (and search), it is just not usable at all
- SCP-4558 pages and any page of the like

I'm aware the second and probably third points means we will have to code something in zimit / warc2zim to handle the situation, which could take months or even years. But we should not lover our quality gates anytime we hit them, and it is not a surprise zimit is not good at handling such a huge website.

Forum is probably not going to be doable, there are just too many pages.

I'm also very intrigued by the final ZIM size. 27G for 10k SCP means 2.7M per SCP. While I do understand some have picture, it seems to be an extraordinarily large ZIM.

**Comment by shadow-absorber** [comment]
> I don't think this ZIM passes Kiwix quality expectations.
> 
> From my PoV, we need to fix:
> 
>     * the illustration
> 
>     * the random URL functionality (and search), it is just not usable at all
> 
>     * SCP-4558 pages and any page of the like
> 
> 
> I'm aware the second and probably third points means we will have to code something in zimit / warc2zim to handle the situation, which could take months or even years. But we should not lover our quality gates anytime we hit them, and it is not a surprise zimit is not good at handling such a huge website.
> 
> Forum is probably not going to be doable, there are just too many pages.
> 
> I'm also very intrigued by the final ZIM size. 27G for 10k SCP means 2.7M per SCP. While I do understand some have picture, it seems to be an extraordinarily large ZIM.

at least this is a lot better then nothing for when someone wants to read some scp:s offline on long trips into the wilderness.
but agree that the remaining problems need something to be figured out

**Comment by IMayBeABitShy** [comment]
> I'm also very intrigued by the final ZIM size. 27G for 10k SCP means 2.7M per SCP. While I do understand some have picture, it seems to be an extraordinarily large ZIM.

It appears that there are a lot of image files included. Using `pyzim-analysis`, we get the following analysis results:

```
===== Overall =====
Counts:
 -> Entries: 157444
 -> Clusters: 10542
 -> Mimetypes: 42
Total size: 28.26 GiB
 -> Entries: 19.78 MiB (0.07%)
 -> Clusters: 28.24 GiB (99.93%)
Compressed Size: 26.68 GiB (94.42%)
==== Used compression types ===
NONE: 9605 (91.11%)
ZSTD: 937 (8.89%)
UNKNOWN: 1 (0.01%)
==== Entry count by mimetype ===
text/html: 46817 (29.74%)
image/png: 23825 (15.13%)
application/json: 21967 (13.95%)
image/jpeg: 20600 (13.08%)
text/css: 1485 (0.94%)
font/woff2: 1141 (0.72%)
image/svg+xml: 1066 (0.68%)
image/gif: 727 (0.46%)
audio/mpeg: 303 (0.19%)
video/mp4: 221 (0.14%)
image/webp: 211 (0.13%)
application/octet-stream: 202 (0.13%)
application/javascript: 158 (0.10%)
text/javascript: 130 (0.08%)
application/font-woff2: 69 (0.04%)
application/x-javascript: 39 (0.02%)
audio/mp4: 39 (0.02%)
image/jpg: 28 (0.02%)
PNG32: 25 (0.02%)
text/plain: 23 (0.01%)
==== Entry size by mimetype ===
application/json: 6.84 MiB (34.57%)
text/html: 4.34 MiB (21.96%)
<redirect>: 3.39 MiB (17.12%)
image/jpeg: 2.59 MiB (13.08%)
image/png: 2.05 MiB (10.34%)
text/css: 121.55 KiB (0.60%)
font/woff2: 112.09 KiB (0.55%)
image/svg+xml: 92.30 KiB (0.46%)
video/mp4: 83.90 KiB (0.41%)
image/gif: 63.46 KiB (0.31%)
audio/mpeg: 24.96 KiB (0.12%)
image/webp: 17.43 KiB (0.09%)
application/octet-stream: 16.94 KiB (0.08%)
application/javascript: 12.42 KiB (0.06%)
text/javascript: 11.56 KiB (0.06%)
application/font-woff2: 9.85 KiB (0.05%)
audio/mp4: 5.62 KiB (0.03%)
image/jpg: 3.69 KiB (0.02%)
application/x-javascript: 3.28 KiB (0.02%)
PNG32: 2.05 KiB (0.01%)
==== Content size by mimetype ===
image/png: 11.54 GiB (40.86%)
image/jpeg: 8.53 GiB (30.22%)
audio/mp4: 1.97 GiB (6.99%)
image/gif: 1.96 GiB (6.94%)
text/html: 1.61 GiB (5.70%)
audio/mpeg: 1006.07 MiB (3.48%)
video/mp4: 871.05 MiB (3.01%)
audio/x-wav: 194.50 MiB (0.67%)
application/x-zip-compressed: 108.99 MiB (0.38%)
application/octet-stream+xapian: 66.36 MiB (0.23%)
text/css: 57.19 MiB (0.20%)
video/quicktime: 57.01 MiB (0.20%)
application/octet-stream: 49.52 MiB (0.17%)
application/ogg: 38.30 MiB (0.13%)
PNG32: 36.58 MiB (0.13%)
font/woff2: 36.06 MiB (0.12%)
image/svg+xml: 33.25 MiB (0.11%)
application/javascript: 24.23 MiB (0.08%)
application/json: 22.27 MiB (0.08%)
text/javascript: 22.06 MiB (0.08%)
==== Entry count by mainpath ===
www.wikidot.com: 30221 (19.19%)
scp-wiki.wikidot.com: 29452 (18.71%)
scp-wiki.wdfiles.com: 28128 (17.87%)
interwiki.scpwiki.com: 21584 (13.71%)
api.crom.avn.sh: 21389 (13.59%)
scp-sandbox-3.wdfiles.com: 2890 (1.84%)
d2qhngyckgiutd.cloudfront.net: 2709 (1.72%)
i.imgur.com: 2531 (1.61%)
images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com: 2236 (1.42%)
upload.wikimedia.org: 1003 (0.64%)
scp-sandbox-3.wikidot.com: 980 (0.62%)
fonts.gstatic.com: 959 (0.61%)
fonts.googleapis.com: 920 (0.58%)
scpsandboxcn.wdfiles.com: 844 (0.54%)
scpsandboxcn.wikidot.com: 769 (0.49%)
66.media.tumblr.com: 573 (0.36%)
64.media.tumblr.com: 555 (0.35%)
scpdsandbox.wdfiles.com: 539 (0.34%)
youtube.fuzzy.replayweb.page: 356 (0.23%)
i.postimg.cc: 291 (0.18%)
==== Entry size by mainpath ===
api.crom.avn.sh: 6.34 MiB (32.07%)
www.wikidot.com: 2.65 MiB (13.39%)
scp-wiki.wikidot.com: 2.42 MiB (12.23%)
scp-wiki.wdfiles.com: 2.36 MiB (11.94%)
interwiki.scpwiki.com: 2.19 MiB (11.06%)
images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com: 1.33 MiB (6.71%)
www.youtube.com: 270.64 KiB (1.34%)
scp-sandbox-3.wdfiles.com: 249.21 KiB (1.23%)
d2qhngyckgiutd.cloudfront.net: 211.82 KiB (1.05%)
gue1-spclient.spotify.com: 134.09 KiB (0.66%)
upload.wikimedia.org: 123.49 KiB (0.61%)
i.imgur.com: 102.64 KiB (0.51%)
fonts.gstatic.com: 97.40 KiB (0.48%)
vod-adaptive-ak.vimeocdn.com: 83.04 KiB (0.41%)
scp-sandbox-3.wikidot.com: 81.46 KiB (0.40%)
fonts.googleapis.com: 79.43 KiB (0.39%)
64.media.tumblr.com: 73.64 KiB (0.36%)
66.media.tumblr.com: 73.23 KiB (0.36%)
scpsandboxcn.wdfiles.com: 65.21 KiB (0.32%)
dwt.soundcloud.com: 64.95 KiB (0.32%)
==== Content size by mainpath ===
scp-wiki.wdfiles.com: 12.15 GiB (43.01%)
i.imgur.com: 2.69 GiB (9.54%)
scp-wiki.wikidot.com: 2.33 GiB (8.26%)
anchor.fm: 2.06 GiB (7.31%)
scp-sandbox-3.wdfiles.com: 1.77 GiB (6.28%)
upload.wikimedia.org: 1.36 GiB (4.81%)
fondationscp.wdfiles.com: 700.24 MiB (2.42%)
images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com: 654.81 MiB (2.26%)
scpsandboxcn.wdfiles.com: 512.49 MiB (1.77%)
youtube.fuzzy.replayweb.page: 432.54 MiB (1.50%)
scpdsandbox.wdfiles.com: 217.57 MiB (0.75%)
64.media.tumblr.com: 216.55 MiB (0.75%)
smlt.wdfiles.com: 195.15 MiB (0.67%)
scp-jp-storage.wdfiles.com: 172.89 MiB (0.60%)
scpsandbox2.wdfiles.com: 172.73 MiB (0.60%)
scpko.wdfiles.com: 140.67 MiB (0.49%)
olicus.wdfiles.com: 133.92 MiB (0.46%)
scp-yerko-6.wdfiles.com: 121.40 MiB (0.42%)
sailorenoch.github.io: 116.27 MiB (0.40%)
scp-sandbox-3.wikidot.com: 110.68 MiB (0.38%)
==== Entries per cluster ===
Min:  1
Max:  38251
Mean: 14.93351038603813
Median: 2
=== Mimetypes per cluster ===
0:     1
1:     6341
2:     2919
3:     968
4:     270
5:     35
6:     6
7:     3
Min:  0
Max:  7
Mean: 1.555060229536185
Median: 1
```

Some observations:
 - 91.11% of clusters are uncompressed - this seems extremely suboptimal
 - the amount of html pages seems reasonable at 47k, but I am surprised that we also got more than 40k images
 - the total size of html pages amounts to 1.61GiB, which should amount to 35KiB per page. Quite large IMO, but not surprisingly large
 -  it does not appear like a lot of unrelated files were included
 - we've got a lot of entries from an API
 - the clusters appear to often contain blobs of mixed mimetypes. This is quite suboptimal for compression.

**Comment by benoit74** [comment]
> at least this is a lot better then nothing for when someone wants to read some scp:s offline on long trips into the wilderness.
but agree that the remaining problems need something to be figured out

This is not Kiwix policy ATM, it has been decided that we do not want to deceive users with low quality stuff (and be bothered by negative feedback about this or that not working). I do get the "better than nothing" argument, it is just hard to convey to end-users. 

> Using pyzim-analysis, we get the following analysis results:

Thank you! I missed this tool, is it another one of your creation or do I still miss some Kiwix tools?

**Comment by IMayBeABitShy** [comment]
> Thank you! I missed this tool, is it another one of your creation or do I still miss some Kiwix tools?

It's one of mine, but quite primitive. It's part of `pyzim-tools`, but I basically just included it because there might be some uses to it. It's somewhat inconsistent in how redirects are handled, though. IIRC the tool counts redirects for some counts but not for others, leading to some wrong percentage. That's probably also the reason that the tool always list some cluster as having unknown compression - it's actually just the redirects.

**Comment by benoit74** [comment]
I like that tool even if primitive, it helps to better grasp ZIM content at a glance

**Comment by observeroftime02** [comment]
While the most recent file / run has produced something that is "good enough" for "my purposes", I do not know if the quality passes the standards set out by the people in charge of administrating this project. There's some definite breakage, and I noticed some more yesterday.

Have a look at SCP-9001 and scroll all the way to the bottom, then click on "SCP-9001.doc". This links out to https://scp-wiki.wikidot.com/scp-9001/offset/1, for example, and I have found a few more instances of such behaviour. I don't know if this is related to the ListPages situation, but there's a few more in the 9000 series that exhibit similar behaviour. SCP-9003, same issue. Clicking the little "claim now" button links out to https://scp-wiki.wikidot.com/scp-9003/offset/1. SCP-9005, SCP-9008, SCP-9034, etc. same story. I could list them all, but don't know if that would be worth the effort. 

I would hesitate to advertise this as a complete scrape of the wiki. It's a partial scrape, a good partial scrape, the best I have seen so far, but there's a lot of room for improvement still. Having all of the main SCP articles / Hubs working and no longer linking out to external resources should be the goal imho.

**Comment by LTVA1** [comment]
All the have `offset/xxx` in the link use ListPages @observeroftime02

**Comment by henk717** [comment]
Are the zim's from this thread available anywhere? I tried running the docker locally but it takes a brutally long time and something crashed my docker instance over half way in after 10 hours of downloading.

**Comment by benoit74** [comment]
This is still not possible to ZIM, tooling needs to be fixed and/or enhanced.
